persist/fs/seek.go

```diff
@@ -262,7 +262,6 @@ func (s *seeker) readIndex(size int) error {
 		// NB(r): Free the unread buffer and reset the decoder as unless
 		// using this seeker in the seeker manager we never use this buffer again
 		s.unreadBuf = nil
-		s.decoder = nil
```

Comment:
In that case can you call `s.decoder.Reset(nil)` instead? Otherwise the decoder holds on to the underlying byte slice which could be a significant chunk of memory